src/vs/workbench/contrib/notebook/common/notebookCommon.ts

```diff
@@ -617,10 +617,31 @@ export namespace CellUri {
 		return parseUri(cell);
 	}
 
-	export function generateCellOutputUri(notebook: URI, outputId?: string) {
+	/**
+	 * Generates a URI for a cell output in a notebook using the output ID.
+	 * Used when URI should be opened as text in the editor.
+	 */
+	export function generateCellOutputUriWithId(notebook: URI, outputId?: string) {
+		return notebook.with({
+			scheme: Schemas.vscodeNotebookCellOutput,
+			query: new URLSearchParams({
+				openIn: 'editor',
+				outputId: outputId || '',
+			}).toString()
+		});
+	}
+	/**
+	 * Generates a URI for a cell output in a notebook using the output index.
+	 * Used when URI should be opened in notebook editor.
+	 */
+	export function generateCellOutputUriWithIndex(notebook: URI, cellUri: URI, outputIndex: number): URI {
 		return notebook.with({
 			scheme: Schemas.vscodeNotebookCellOutput,
-			fragment: `op${outputId ?? ''},${notebook.scheme !== Schemas.file ? notebook.scheme : ''}`
```

Comment:
need to fix this since im not correctly handling`${notebook.scheme !== Schemas.file ? notebook.scheme : ''}`. Not entirely sure what notebook.scheme other than file since the only entry point I found was the "open in a text editor" and when I tried it the value was type "file". Either I don't understand the use case or this is redundant and I can actually remove the scheme from the Uri to simplify

Reply:
The scheme might be different in remote cases, or if the user just happens to have some custom file provider, but the untitled scheme is also used if in an unsaved notebook